From c6c7068e995c00d978282e7103c04ffcffca9a23 Mon Sep 17 00:00:00 2001
From: Nick Wellnhofer <wellnhofer@aevum.de>
Date: Mon, 21 Nov 2022 22:09:19 +0100
Subject: [PATCH 24/28] parser: Remove dangerous check in xmlParseCharData

If this check succeeds, xmlParseCharData could be called over and over
again without making progress, resulting in an infinite loop.

It's only important to check for XML_PARSER_EOF which is done later.

Related to #441.

Reference: https://github.com/GNOME/libxml2/commit/0e193f0d61f6d6f29c31ac5f801975e810df7a04
Conflict: NA
---
 parser.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/parser.c b/parser.c
index 4360479..4405a7e 100644
--- a/parser.c
+++ b/parser.c
@@ -4535,9 +4535,6 @@ get_more:
                     line = ctxt->input->line;
                     col = ctxt->input->col;
 		}
-                /* something really bad happened in the SAX callback */
-                if (ctxt->instate != XML_PARSER_CONTENT)
-                    return;
 	    }
 	    ctxt->input->cur = in;
 	    if (*in == 0xD) {
-- 
2.27.0

